### ISSUE TITLE
Allow extending the TileData struct

### ISSUE THREAD

**Issue by duronrulez** [Unity-Technologies/2d-extras#75]
Being able to extend the TileData struct would be helpful when you want to add custom data to each tile.
I can see that being a sealed class, so if extending is not an option, having something like a Dictionary or something similar inside the TileData struct would also work.

If im completely mistaken and there is another way to have each tile contain their own set of data, please let me know.

**Comment by edwardrowe** [comment]
I haven't used it, but there's GridInformation which I think is what you
want. Allows you to add data to tile positions using a String key + Value.

https://github.com/Unity-Technologies/2d-extras/blob/master/Assets/Tilemap/GridInformation/Scripts/GridInformation.cs

**Comment by duronrulez** [comment]
Yes i've seen that script, it does kind of what its needed. But it seems fiddlier than needed, considering you have the tiledata readily available in the tile itself as well.

Right now if you want to lets say display different sprite based on some value, you have to instantiate that component somewhere, get the gameobject for it and then place and read your values to / from it.

Seems a bit too convoluted, but yes it does work :)
